cmd/provider/main.go

```diff
@@ -40,6 +40,7 @@ import (
 	scv1alpha1 "github.com/crossplane-contrib/provider-gcp/apis/v1alpha1"
 	gcp "github.com/crossplane-contrib/provider-gcp/pkg/controller"
 	"github.com/crossplane-contrib/provider-gcp/pkg/features"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
```

Comment:
WIthout this import, running `make run` failed for me. I believe this is due to google changing the authentication mechanism for GKE. After some research, I found that other people used this solution to get their controllers to work while testing outside of the cluster.

Reply:
For the tutorial comment, I agree we should specify the requirements for GKE. I think this solution will only work if your controlplane is running in GKE with workload identity. Otherwise, the CronJob would need some sort of GCP service account key I assume which kind of defeats the whole purpose.  I will work on some disclaimers in the documentation and push another commit to this PR.